extensions/azure-keyvault/runtime/src/main/java/io/quarkiverse/azure/keyvault/secret/runtime/config/KeyVaultSecretConfigUtil.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkiverse.azure.keyvault.secret.runtime.config;
+
+import com.azure.security.keyvault.secrets.models.KeyVaultSecretIdentifier;
+
+public class KeyVaultSecretConfigUtil {
+    private static final String AZURE_KEYVAULT_PREFIX = "kv//";
+    private static final String AZURE_KEYVAULT_ENDPOINT_PREFIX = "https://";
+    private static final String AZURE_VAULT_URL_FORMAT = "https://%s.vault.azure.net/secrets/%s/%s";
+
+    private KeyVaultSecretConfigUtil() {
+    }
+
+    static KeyVaultSecretIdentifier getSecretIdentifier(String input, String defaultEndpoint) {
+
+        if (!input.startsWith(AZURE_KEYVAULT_PREFIX)) {
+            return null;
+        }
+
+        String resourcePath = input.substring(AZURE_KEYVAULT_PREFIX.length());
+        String[] tokens = resourcePath.split("/");
```

Comment:
I don't see you checking that the prefix is actually `kv`. It looks like you're just assuming the first two letters are `kv`. I think this should be checked.